### PR TITLE
introduce benchmark overhead

### PR DESCRIPTION
**Pull request by mordamax**
Fix: https://github.com/paritytech/command-bot/issues/102
introduce benchmark overhead 
separate parsing bench-bot command from main runner so it's easier to test cargo run output

```
# cumulus
... bench-bot $ overhead assets statemint
cargo run --locked --quiet --profile=production -- benchmark overhead --execution=wasm --wasm-execution=compiled --warmup=10 --repeat=100 --weight-path=./cumulus/parachains/runtimes/assets/statemint/src/weights --chain=statemint

# substrate
... bench-bot $ overhead
cargo run --locked --quiet --profile=production -- benchmark overhead --execution=wasm --wasm-execution=compiled --warmup=10 --repeat=100 --weight-path=./frame/support/src/weights --chain=dev

# polkadot
...bench-bot $ overhead westend
cargo run --locked --quiet --profile=production -- benchmark overhead --execution=wasm --wasm-execution=compiled --warmup=10 --repeat=100 --weight-path=./runtime/westend/constants/src/weights --chain=westend-dev
```

Testing 
- https://github.com/tripleightech/cumulus/pull/1
- https://github.com/tripleightech/polkadot/pull/1

### CODE REVIEW

**Review by alvicsam — APPROVED** [review]

File: bench-bot-run.sh [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+# This file is separated to simplify testing the final pure command output
+
+set -eu -o pipefail
+shopt -s inherit_errexit
+
+. "$(dirname "${BASH_SOURCE[0]}")/utils.sh"
+. "$(dirname "${BASH_SOURCE[0]}")/cmd_runner.sh"
+
+cargo_run_benchmarks="cargo run --locked --quiet --profile=production"
```
Comment (mordamax) [inline bench-bot-run.sh:11]: removed `+nightly`

File: bench-bot-run.sh [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  pallet
+  --steps=50
+  --repeat=20
+  --extrinsic="*"
+  --execution=wasm
+  --wasm-execution=compiled
+  --heap-pages=4096
+  --json-file="${ARTIFACTS_DIR}/bench.json"
+)
+bench_pallet() {
```
Comment (mordamax) [inline bench-bot-run.sh:37]: this part remained untouched

File: bench-bot-run.sh [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+bench_overhead_common_args=(
+  --
+  benchmark
+  overhead
+  --execution=wasm
+  --wasm-execution=compiled
+  --warmup=10
+  --repeat=100
+)
+bench_overhead() {
```
Comment (mordamax) [inline bench-bot-run.sh:170]: this is new one

**Review by ggwpez — APPROVED** [review]
Works locally.  
Did the testing with the bot work? I cannot login to the testing GitLab.

**Pull request merged**